### ISSUE TITLE
[BUG]Raising the skill of a custom skill throws NRE

### ISSUE THREAD

**Issue by paddywaan** [Valheim-Modding/Jotunn#85]
RaiseSkill throws NRE when given a custom skill, as such:
```cs
        void addSkills()
        {
            // Test adding a skill with a texture
            Sprite testSkillSprite = Sprite.Create(testTex, new Rect(0f, 0f, testTex.width, testTex.height), Vector2.zero);
            //TestSkillType = SkillManager.Instance.AddSkill("com.jotunnlib.JotunnModExample.testskill", "TestingSkill", "A nice testing skill!", 1f, testSkillSprite);
            TestSkillType = SkillManager.Instance.AddSkill(new SkillConfig { Identifier = "com.jotunnlib.JotunnModExample.testskill",
                Name = "TestingSkill",
                Description = "A nice testing skill!",
                Icon = testSkillSprite,
                IncreaseStep = 1f
            }, true);
            Logger.LogDebug(TestSkillType);
            //if(!TestSkillType) Logger.
        }

private void Update()
        {
            if (Input.GetKeyDown(KeyCode.F8))
            { // Set a breakpoint here to break on F6 key press
                Player.m_localPlayer.RaiseSkill(TestSkillType, 1);
            }
        }
```

The first stack trace occurs on hitting F8. The second occurs upon opening the skills window.
```
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
Skills+Skill.Raise (System.Single factor) (at <6fd2223d69c147d18c77057ce0dedd7b>:0)
Skills.RaiseSkill (Skills+SkillType skillType, System.Single factor) (at <6fd2223d69c147d18c77057ce0dedd7b>:0)
Player.RaiseSkill (Skills+SkillType skill, System.Single value) (at <6fd2223d69c147d18c77057ce0dedd7b>:0)
JotunnModExample.JotunnModExample.Update ()

[Info   : Unity Log] 04/11/2021 23:21:19: Setting selected recipe 0

[Info   : Unity Log] 04/11/2021 23:21:19: UI Group status changed root = True

[Info   : Unity Log] 04/11/2021 23:21:19: UI Group status changed Player = True

[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
SkillsDialog.Setup (Player player) (at <6fd2223d69c147d18c77057ce0dedd7b>:0)
InventoryGui.OnOpenSkills () (at <6fd2223d69c147d18c77057ce0dedd7b>:0)
UnityEngine.Events.InvokableCall.Invoke () (at <51725b129ecd4967a9a213c1ae65b295>:0)
UnityEngine.Events.UnityEvent.Invoke () (at <51725b129ecd4967a9a213c1ae65b295>:0)
UnityEngine.UI.Button.Press () (at <5abdc278a459476dabe36f8ad65ea091>:0)
UnityEngine.UI.Button.OnPointerClick (UnityEngine.EventSystems.PointerEventData eventData) (at <5abdc278a459476dabe36f8ad65ea091>:0)
UnityEngine.EventSystems.ExecuteEvents.Execute (UnityEngine.EventSystems.IPointerClickHandler handler, UnityEngine.EventSystems.BaseEventData eventData) (at <5abdc278a459476dabe36f8ad65ea091>:0)
UnityEngine.EventSystems.ExecuteEvents.Execute[T] (UnityEngine.GameObject target, UnityEngine.EventSystems.BaseEventData eventData, UnityEngine.EventSystems.ExecuteEvents+EventFunction`1[T1] functor) (at <5abdc278a459476dabe36f8ad65ea091>:0)
UnityEngine.EventSystems.EventSystem:Update()
```

When raise skill is performed, the sword skill is set to level 100: 
![image](https://user-images.githubusercontent.com/70168382/114323606-d8e25300-9b1d-11eb-9034-233f651cd633.png)

**Comment by raduschirliu** [comment]
Did this still happen after PR #16 was merged in?

**Comment by paddywaan** [comment]
Before. Did you have this working and tested in JL? The raising of a custom skill I mean?

**Comment by raduschirliu** [comment]
From what I remember, there were some issues with raising custom skills in JL. I think most of those should have been fixed with the PR I mentioned though, since a lot of the functionality to do with skills & their patches was re-written

**Comment by paddywaan** [comment]
resolved by #16 

![image](https://user-images.githubusercontent.com/70168382/114501182-60fd5100-9c21-11eb-9838-b7c13569699d.png)
